xcb-types.el

```diff
@@ -731,7 +730,8 @@ and the second the consumed length."
                                                (eieio-oref-default obj name)
                                                nil total-length))
                (setf (slot-value obj name) (car tmp))
-               (setq count (+ count (cadr tmp)))
+               (setq count (+ count (cadr tmp))
+                     offset (+ offset (cadr tmp)))
                (setq data (substring data (cadr tmp)))))))
```

Comment:
The allocation is still there.

Reply:
🤦

Reply:
(I blame IEEE754 for my current brain function)

Reply:
Haha. Ideally Elisp would have some low level primitives for byte packing, such that we don't have to waste our higher brain functions. :-P